providers/azure.go

```diff
@@ -81,7 +81,7 @@ func NewAzureProvider(p *ProviderData, opts options.AzureOptions) *AzureProvider
 	}
 	p.getAuthorizationHeaderFunc = makeAzureHeader
 
-	tenant := "common"
+	var tenant string
```

Comment:
I don't think we want this change, this is where it is being defaulted if we remove the defaulting from the options package

Reply:
yep. I's starting to think that changing the defaults for Azure wasn't such a good idea after all.

Reply:
I think what's missing here is that there are two places we can handle defaults, at the moment, there's a lot in the config. We can also handle it at this level as well. So i think what we want is to remove all azure defaulting from everywhere else in the code, and then `opts.Tenant` should be empty of the user provided value, in which case we can default it to common here if we drop this particular change